sklearn/neighbors/_base.py

```diff
@@ -1249,13 +1250,13 @@ class from an array representing our data set and ask who's
             )
             if return_distance:
                 neigh_dist_chunks, neigh_ind_chunks = zip(*chunked_results)
-                neigh_dist_list = sum(neigh_dist_chunks, [])
-                neigh_ind_list = sum(neigh_ind_chunks, [])
+                neigh_dist_list = reduce(operator.iadd, neigh_dist_chunks, [])
+                neigh_ind_list = reduce(operator.iadd, neigh_ind_chunks, [])
                 neigh_dist = _to_object_array(neigh_dist_list)
                 neigh_ind = _to_object_array(neigh_ind_list)
                 results = neigh_dist, neigh_ind
             else:
-                neigh_ind_list = sum(chunked_results, [])
+                neigh_ind_list = reduce(operator.iadd, chunked_results, [])
```

Comment:
I don't find these ones much better. And I don't think it's the recommended way to do that. I would be okay to replace with
`list(itertools.chain.from_iterable(neigh_dist_chunks))`
What do you think ?

Reply:
Sure. That's probably the most readable of all, at least to me.